sda/cmd/verify/verify.go

```diff
@@ -204,13 +205,45 @@ func main() {
 				continue
 			}
 
-			hr := bytes.NewReader(header)
-			// Feed everything read from the archive file to archiveFileHash
-			mr := io.MultiReader(hr, io.TeeReader(f, archiveFileHash))
+			// var c4ghr *streaming.Crypt4GHReader
+
+			// for _, key := range archiveKeyList {
+			// 	// Feed data read from the archive file into archiveFileHash
+			// 	mr := io.MultiReader(bytes.NewReader(header), io.TeeReader(f, archiveFileHash))
+
+			// 	c4ghr, err = streaming.NewCrypt4GHReader(mr, *key, nil)
+			// 	if (err == nil) && (c4ghr != nil) {
+			// 		break
+			// 	}
+			// 	log.Warnf("Failed to open C4GH decryptor with key: %v, trying next key. Reason: %s", *key, err.Error())
+			// }
+			// if c4ghr == nil {
+			// 	log.Errorf("Failed to open C4GH decryptor stream with all available key(s)")
+
+			// 	continue
+			// }
+			var key *[32]byte
+			var foundKey bool
+			for _, key = range archiveKeyList {
+				size, err := headers.EncryptedSegmentSize(header, *key)
+				if (err == nil) && (size != 0) {
+					foundKey = true
+
+					break
+				}
+				log.Warnf("Failed to get encrypted segment size with key: %v, trying next key. Reason: %s", *key, err.Error())
+			}
+
+			if !foundKey {
+				log.Errorf("Failed to get encrypted segment size with all available key(s).")
+
+				continue
+			}
 
+			mr := io.MultiReader(bytes.NewReader(header), io.TeeReader(f, archiveFileHash))
 			c4ghr, err := streaming.NewCrypt4GHReader(mr, *key, nil)
 			if err != nil {
-				log.Errorf("Failed to open c4gh decryptor stream, reson: (%s)", err.Error())
+				log.Errorf("Failed to open C4GH decryptor stream with key: %v", *key)
```

Comment:
```suggestion
				log.Errorf("failed to open c4gh decryptor stream, reson: (%s)", err.Error())
```
Do NOT leak the crypt4gh key!!